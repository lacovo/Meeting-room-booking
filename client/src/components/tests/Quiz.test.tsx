import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Quiz from '../Quiz';

configure({ adapter: new Adapter() });

jest.mock('react-redux', () => ({
  useDispatch: () => {},
  useSelector: () => {
    return [{ question: 'Test' }];
  },
}));

jest.mock('react-router', () => ({
  useParams: () => {
    return { questionIndex: '1' };
  },
  useHistory: () => {},
  Route: () => {},
}));

describe('<Quiz />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const wrapper = shallow(<Quiz />);
    expect(wrapper).not.toBeNull();
    expect(spy).not.toHaveBeenCalled();
  });
  it('Should render and match the snapshot', () => {
    const wrapper = shallow(<Quiz />);
    expect(wrapper).toMatchSnapshot();
  });
});
