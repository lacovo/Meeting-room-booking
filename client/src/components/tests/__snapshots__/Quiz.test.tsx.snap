// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Quiz /> Should render and match the snapshot 1`] = `ShallowWrapper {}`;
